detection-rules/link_squarespace_abuse.yml

```
@@ -0,0 +1,28 @@
+name: "Link: Squarespace Infrastructure Abuse"
+description: "Detects inbound messages containing exactly one Squarespace tracking link but lacking authentic Squarespace email headers and sender patterns."
+type: "rule"
+severity: "medium"
+source: |
+  type.inbound
+  and any(body.links, .href_url.domain.domain == "engage.squarespace-mail.com")
+  // there is one unique Squarespace Link in the message
+  and length(distinct(filter(body.links,
+                             .href_url.domain.domain == "engage.squarespace-mail.com"
+                      ),
+                      .href_url.url
+             )
+  ) == 1
+  and not headers.return_path.domain.root_domain == "squarespace-mail.com"
```

https://platform.sublime.security/messages/21489b61ebea27c1d819732676561828ef73140c26a8aeaf84f811c944c787ac?preview_id=0195d924-ceca-704d-a7fb-1c18ca546e3c
```suggestion
  and not headers.return_path.domain.root_domain == "squarespace-mail.com"
  and not any(headers.domains, .root_domain == "squarespace-mail.com")
```